### PR TITLE
[SMALLFIX] Add an option to disable netty channel pool

### PR DESCRIPTION
**Pull request by peisun1115**
This is to provide an option to make >=1.3.0 client backward compatible with <=1.2.0 server

### CODE REVIEW

**Comment by calvinjia** [comment]
LGTM

File: docs/_data/table/cn/user-configuration.yml [hunk 1]
```diff
@@ -49,6 +49,9 @@ alluxio.user.network.netty.channel.pool.size.max:
   netty网络通道池的最大容量。
 alluxio.user.network.netty.channel.pool.gc.threshold.ms:
   netty网络通道会被关闭如果它被闲置超过这个时间。
+alluxio.user.network.netty.channel.pool.disable:
+  禁用netty网络通道池特性。设置这个选项为真如果客户端的版本 >= 1.3.0 但是服务器版本 <= 1.2.0。
+  Disable netty channel pool. This should be turned on if > 1.3.0 client is used with <= 1.2.0 server.
```
Comment (ifcharming) [inline docs/_data/table/cn/user-configuration.yml]: nit: >=1.3.0
<= 1.2.x server

File: docs/_data/table/cn/user-configuration.yml [hunk 2]
```diff
@@ -49,6 +49,9 @@ alluxio.user.network.netty.channel.pool.size.max:
   netty网络通道池的最大容量。
 alluxio.user.network.netty.channel.pool.gc.threshold.ms:
   netty网络通道会被关闭如果它被闲置超过这个时间。
+alluxio.user.network.netty.channel.pool.disable:
+  禁用netty网络通道池特性。设置这个选项为真如果客户端的版本 >= 1.3.0 但是服务器版本 <= 1.2.0。
```
Comment (ifcharming) [inline docs/_data/table/cn/user-configuration.yml]: <=1.2.x

File: docs/_data/table/en/user-configuration.yml [hunk 3]
```diff
@@ -61,6 +61,8 @@ alluxio.user.network.netty.channel.pool.size.max:
   The maximum number of netty channels cached in the netty channel pool.
 alluxio.user.network.netty.channel.pool.gc.threshold.ms:
   A netty channel is closed if it has been idle for more than this threshold.
+alluxio.user.network.netty.channel.pool.disable:
+  Disable netty channel pool. This should be turned on if the client version is >= 1.3.0 but server version is <= 1.2.0.
```
Comment (ifcharming) [inline docs/_data/table/en/user-configuration.yml]: <=1.2.x

**Comment by peisun1115** [comment]
@ifcharming It is not possible to distinguish closed channel due to version mismatch because the channel is always closed after every request for versions <=1.2.0.

File: docs/_data/table/cn/user-configuration.yml (hunk 2 above)
Comment (peisun1115) [inline docs/_data/table/cn/user-configuration.yml]: done

File: docs/_data/table/cn/user-configuration.yml (hunk 1 above)
Comment (peisun1115) [inline docs/_data/table/cn/user-configuration.yml]: done

File: docs/_data/table/en/user-configuration.yml (hunk 3 above)
Comment (peisun1115) [inline docs/_data/table/en/user-configuration.yml]: done

**Comment by peisun1115** [comment]
@ifcharming fixed. thanks

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 4]
```diff
@@ -241,6 +241,7 @@
   USER_NETWORK_NETTY_CHANNEL_POOL_SIZE_MAX(Name.USER_NETWORK_NETTY_CHANNEL_POOL_SIZE_MAX, 1024),
   USER_NETWORK_NETTY_CHANNEL_POOL_GC_THRESHOLD_MS(
       Name.USER_NETWORK_NETTY_CHANNEL_POOL_GC_THRESHOLD_MS, 300 * Constants.SECOND_MS),
+  USER_NETWORK_NETTY_CHANNEL_POOL_DISABLE(Name.USER_NETWORK_NETTY_CHANNEL_POOL_DISABLE, false),
```
Comment (ifcharming) [inline core/common/src/main/java/alluxio/PropertyKey.java]: nit: is Alluxio naming convention `DISABLED` instead of `DISABLE` ?

Comment (peisun1115) [inline core/common/src/main/java/alluxio/PropertyKey.java]: done

**Comment by peisun1115** [comment]
@ifcharming thanks. i have renamed it.

**Comment by ifcharming** [comment]
LGTM, thanks for the fix!

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
@ifcharming 

The failed tests don't seem to be related. 

recoverClusterSuccess(alluxio.security.ClusterInitializationTest)  Time elapsed: 0.089 sec  <<< ERROR!
java.lang.RuntimeException: alluxio.exception.AccessControlException: Permission denied: Unauthorized user on root
	at alluxio.master.file.meta.InodeTree.addInodeFromJournal(InodeTree.java:840)
	at alluxio.master.file.FileSystemMaster.processJournalEntry(FileSystemMaster.java:299)
	at alluxio.master.AbstractMaster.processJournalCheckpoint(AbstractMaster.java:82)
	at alluxio.master.journal.JournalTailer.processJournalCheckpoint(JournalTailer.java:98)
	at alluxio.master.AbstractMaster.start(AbstractMaster.java:136)
	at alluxio.master.file.FileSystemMaster.start(FileSystemMaster.java:401)
	at alluxio.master.MasterTestUtils.createLeaderFileSystemMasterFromJournal(MasterTestUtils.java:33)
	at alluxio.security.ClusterInitializationTest.recoverClusterSuccess(ClusterInitializationTest.java:78

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by ifcharming** [comment]
@peisun1115  you can ignore the flakiness.

**Comment by peisun1115** [comment]
Jenkins, test this please

**Comment by peisun1115** [comment]
@ifcharming @calvinjia can you help to merge? thanks

**Pull request merged**